test/functional/p2p_blocksonly.py

```diff
@@ -55,8 +55,51 @@ def run_test(self):
         with self.nodes[0].assert_debug_log(['received getdata for: tx {} peer=1'.format(txid)]):
             self.nodes[0].sendrawtransaction(sigtx)
             self.nodes[0].p2p.wait_for_tx(txid)
+            pprint(self.nodes[0].p2p.wait_for_tx(txid))
             assert_equal(self.nodes[0].getmempoolinfo()['size'], 1)
 
+        self.log.info("Restarting node 0 with whitelist permission and blocksonly")
+        self.stop_node(0)
+        self.start_node(0, ["-whitelist=127.0.0.1", "-whitelistforcerelay", "-blocksonly"])
+        self.nodes[0].add_p2p_connection(P2PInterface())
+        self.log.info('Check that txs from whitelisted peers are not rejected and relayed to others')
+        peerInfo = self.nodes[0].getpeerinfo()[0]
+        assert_equal(peerInfo['whitelisted'], True)
+        assert_equal(peerInfo['permissions'], ['noban', 'forcerelay', 'relay', 'mempool']) #checkpermission use check permission from p2p_permission
+
+        # create a new transaction
+        prevtx1 = self.nodes[0].getblock(self.nodes[0].getblockhash(2), 2)['tx'][0]
+        rawtx1 = self.nodes[0].createrawtransaction(
+            inputs=[{
+                'txid': prevtx1['txid'],
+                'vout': 0
+            }],
+            outputs=[{
+                self.nodes[0].get_deterministic_priv_key().address: 50 - 0.00125
+            }],
+        )
+        sigtx1 = self.nodes[0].signrawtransactionwithkey(
+            hexstring=rawtx1,
+            privkeys=[self.nodes[0].get_deterministic_priv_key().key],
+            prevtxs=[{
+                'txid': prevtx1['txid'],
+                'vout': 0,
+                'scriptPubKey': prevtx1['vout'][0]['scriptPubKey']['hex'],
+            }],
+        )['hex']
+        txid1 = self.nodes[0].testmempoolaccept([sigtx1])[0]['txid']
+        self.nodes[0].p2p.send_message(msg_tx(FromHex(CTransaction(), sigtx1))) # so self is the transaction sender?
+
+        pprint(self.nodes[0].p2p.last_message)
+        pprint(self.nodes[0].p2p.last_message.get('tx'))
+
+        with self.nodes[0].assert_debug_log(['received getdata for: tx {} peer=0'.format(txid1)]):
+            self.nodes[0].sendrawtransaction(sigtx1)
+            self.nodes[0].p2p.wait_for_tx(txid1)
+            pprint(self.nodes[0].p2p.wait_for_tx(txid1))
```

Comment:
I am getting this error from test: 
`File "/Users/gwang/Documents/bitcoin/test/functional/test_framework/util.py", line 234, in wait_until
    raise AssertionError("Predicate {} not true after {} seconds".format(predicate_source, timeout))
AssertionError: Predicate ''''
        def test_function():
            assert self.is_connected
            if not self.last_message.get('tx'):
                return False
            return self.last_message['tx'].tx.rehash() == txid
''' not true after 60 seconds`

I'm a bit confused about why this transaction isn't working, since I constructed it very similarly to the first transaction in this test file. I am still learning the way that this test setup works, and I might have been missing something